service/src/LevelZero.cpp

```diff
@@ -508,48 +536,99 @@ namespace geopm
         return {result_active, result_timestamp};
     }
 
-    uint64_t LevelZeroImp::energy_timestamp(unsigned int l0_device_idx) const
+    uint64_t LevelZeroImp::energy_timestamp(int geopm_domain,
+                                            unsigned int l0_device_idx,
+                                            int l0_domain,
+                                            int l0_domain_idx) const
     {
-        return m_devices.at(l0_device_idx).cached_energy_timestamp;
+        uint64_t timestamp = 0;
+        //PACKAGE
+        if(geopm_domain == GEOPM_DOMAIN_GPU) {
+            timestamp = m_devices.at(l0_device_idx).cached_energy_timestamp;
+        }
+        else if(geopm_domain == GEOPM_DOMAIN_GPU_CHIP) {
+        //TILE
+            timestamp = m_devices.at(l0_device_idx).subdevice.cached_energy_timestamp.at(l0_domain_idx);
+        }
+        //TODO: either check l0_domain, or move to treating it the same
+        //      as all other signals (i.e. All, Compute, and Mem)
```

Comment:
Is this implementation complete?

Reply:
These checks have been moved to the LevelZeroDevicePool.  Latest commit removes the TODOs.